tools/http_message.go

```diff
@@ -0,0 +1,51 @@
+package tools
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// HTTPMessage represents data exchanged between server and client, typed for encryption purposes.
+type HTTPMessage string
+
+// NewHTTPMessage generates the base64-encoded message required to make API-Key-Authenticated requests.
+func NewHTTPMessage(req *http.Request, timestamp string) HTTPMessage {
+	if req.Body == nil {
+		return HTTPMessage([]byte{})
+	}
+
+	body, _ := io.ReadAll(req.Body)
+	req.Body = io.NopCloser(bytes.NewBuffer(body))
```

Comment:
```suggestion
        var body []byte
        if req.Body != nil {
                body, _ := io.ReadAll(req.Body)
                req.Body = io.NopCloser(bytes.NewBuffer(body))
        }
```

I think we only want one return call in this function, to keep the formatting secure.

Reply:
@prestonvasquez okay, I'll fix this, but I need to make a couple more commits because this change breaks the tests. I will notify you.